CodeQL: Speed up Maven build

kohlschuetter opened the pull request:
For CodeQL anaylsis, we can ignore Checkstyle and License goals. Tthese goals are covered the CI phase that runs concurrently.

Local testing indicates a significant build time improvement (2:33 min -> 1:36 min).

Given that the maven phase in GitHub's CodeQL takes around 20 minutes, we may save quite some time.

Signed-off-by: Christian Kohlschütter <christian@kohlschutter.com>

kohlschuetter committed:
CodeQL: Speed up Maven build, use "fast" build profile, use parallelism

For CodeQL anaylsis, we can ignore Checkstyle, License and Javadoc
goals. These goals are covered the CI phase that runs concurrently.

There already is a profile named "fast" disabling some plugins, among
other settings.

Local testing indicates a significant build time improvement (2:33 min
-> 1:36 min single-threaded -> 0:38 min with -T4).

Given that the maven phase in GitHub's CodeQL takes around 20 minutes,
we may save quite some time.

Signed-off-by: Christian Kohlschütter <christian@kohlschutter.com>

olamy commented:
There is a profile called fast for this ;)

joakime commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -80,7 +80,8 @@ jobs:
 
       - name: Clean install dependencies and build
-        run: mvn clean install -DskipTests -B -V
+        run: mvn clean install -DskipTests -B -V -Dcheckstyle.skip=true -Dlicense.skip=true -Dmaven.site.skip=true -Dmaven.javadoc.skip=true
```
How about this way instead ...

```suggestion
        run: mvn clean install -DskipTests -B -V -Pfast
```

joakime reviewed (dismissed):
Try the `fast` profile first.

joakime commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -80,7 +80,7 @@ jobs:
 
       - name: Clean install dependencies and build
-        run: mvn clean install -DskipTests -B -V -Pfast -Dmaven.javadoc.skip=true -T4
+        run: mvn clean install -DskipTests -B -V -Pfast -Dmaven.javadoc.skip=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss,SSS -Dorg.slf4j.simpleLogger.showDateTime=true
```
We should add javadoc.skip to the `fast` profile.

kohlschuetter replied:
Good idea. However, I'm not seeing any improvements on Github's end (still over 17 minutes for that command). Currently debugging to see what's going on

kohlschuetter committed:
pom: "fast" profile: Disable javadoc, Xlint

kohlschuetter committed:
Re-enable javadoc upon -Pfast, disable unnecessary plugins entirely

javadoc assets are required by some other code. Disabling javadoc
generation may cause CI builds to fail.

Disabling plugin execution instead of just setting skip=true may save
RAM, not necessarily speed up things.

kohlschuetter commented:
Abandoning the effort to speed up CodeQL. The CodeQL action is slow because their VM is heavily throttled, not much we can do here.
